{"uid":"4c4e2ed4c9924b5b","name":"test_data_replication","fullName":"tests.integration.high_availability.test_async_replication#test_data_replication","historyId":"6c6a39d1deef0f5c857c63ef8908d129","time":{"start":1719020054223,"stop":1719020062307,"duration":8084},"description":"Test to write to primary, and read the same data back from replicas.","descriptionHtml":"<p>Test to write to primary, and read the same data back from replicas.</p>\n","status":"broken","statusMessage":"mysql.connector.errors.ProgrammingError: 1049 (42000): Unknown database 'continuous_writes_database'","statusTrace":"Traceback (most recent call last):\n  File \"/home/runner/work/mysql-k8s-operator/mysql-k8s-operator/.tox/integration/lib/python3.10/site-packages/mysql/connector/connection_cext.py\", line 611, in cmd_query\n    self._cmysql.query(\n_mysql_connector.MySQLInterfaceError: Unknown database 'continuous_writes_database'\n\nThe above exception was the direct cause of the following exception:\n\nTraceback (most recent call last):\n  File \"/home/runner/work/mysql-k8s-operator/mysql-k8s-operator/.tox/integration/lib/python3.10/site-packages/_pytest/runner.py\", line 341, in from_call\n    result: Optional[TResult] = func()\n  File \"/home/runner/work/mysql-k8s-operator/mysql-k8s-operator/.tox/integration/lib/python3.10/site-packages/_pytest/runner.py\", line 262, in <lambda>\n    lambda: ihook(item=item, **kwds), when=when, reraise=reraise\n  File \"/home/runner/work/mysql-k8s-operator/mysql-k8s-operator/.tox/integration/lib/python3.10/site-packages/pluggy/_hooks.py\", line 513, in __call__\n    return self._hookexec(self.name, self._hookimpls.copy(), kwargs, firstresult)\n  File \"/home/runner/work/mysql-k8s-operator/mysql-k8s-operator/.tox/integration/lib/python3.10/site-packages/pluggy/_manager.py\", line 120, in _hookexec\n    return self._inner_hookexec(hook_name, methods, kwargs, firstresult)\n  File \"/home/runner/work/mysql-k8s-operator/mysql-k8s-operator/.tox/integration/lib/python3.10/site-packages/pluggy/_callers.py\", line 182, in _multicall\n    return outcome.get_result()\n  File \"/home/runner/work/mysql-k8s-operator/mysql-k8s-operator/.tox/integration/lib/python3.10/site-packages/pluggy/_result.py\", line 100, in get_result\n    raise exc.with_traceback(exc.__traceback__)\n  File \"/home/runner/work/mysql-k8s-operator/mysql-k8s-operator/.tox/integration/lib/python3.10/site-packages/pluggy/_callers.py\", line 103, in _multicall\n    res = hook_impl.function(*args)\n  File \"/home/runner/work/mysql-k8s-operator/mysql-k8s-operator/.tox/integration/lib/python3.10/site-packages/_pytest/runner.py\", line 177, in pytest_runtest_call\n    raise e\n  File \"/home/runner/work/mysql-k8s-operator/mysql-k8s-operator/.tox/integration/lib/python3.10/site-packages/_pytest/runner.py\", line 169, in pytest_runtest_call\n    item.runtest()\n  File \"/home/runner/work/mysql-k8s-operator/mysql-k8s-operator/.tox/integration/lib/python3.10/site-packages/_pytest/python.py\", line 1792, in runtest\n    self.ihook.pytest_pyfunc_call(pyfuncitem=self)\n  File \"/home/runner/work/mysql-k8s-operator/mysql-k8s-operator/.tox/integration/lib/python3.10/site-packages/pluggy/_hooks.py\", line 513, in __call__\n    return self._hookexec(self.name, self._hookimpls.copy(), kwargs, firstresult)\n  File \"/home/runner/work/mysql-k8s-operator/mysql-k8s-operator/.tox/integration/lib/python3.10/site-packages/pluggy/_manager.py\", line 120, in _hookexec\n    return self._inner_hookexec(hook_name, methods, kwargs, firstresult)\n  File \"/home/runner/work/mysql-k8s-operator/mysql-k8s-operator/.tox/integration/lib/python3.10/site-packages/pluggy/_callers.py\", line 182, in _multicall\n    return outcome.get_result()\n  File \"/home/runner/work/mysql-k8s-operator/mysql-k8s-operator/.tox/integration/lib/python3.10/site-packages/pluggy/_result.py\", line 100, in get_result\n    raise exc.with_traceback(exc.__traceback__)\n  File \"/home/runner/work/mysql-k8s-operator/mysql-k8s-operator/.tox/integration/lib/python3.10/site-packages/pluggy/_callers.py\", line 103, in _multicall\n    res = hook_impl.function(*args)\n  File \"/home/runner/work/mysql-k8s-operator/mysql-k8s-operator/.tox/integration/lib/python3.10/site-packages/_pytest/python.py\", line 194, in pytest_pyfunc_call\n    result = testfunction(**testargs)\n  File \"/home/runner/work/mysql-k8s-operator/mysql-k8s-operator/.tox/integration/lib/python3.10/site-packages/pytest_asyncio/plugin.py\", line 529, in inner\n    _loop.run_until_complete(task)\n  File \"/usr/lib/python3.10/asyncio/base_events.py\", line 649, in run_until_complete\n    return future.result()\n  File \"/home/runner/work/mysql-k8s-operator/mysql-k8s-operator/tests/integration/high_availability/test_async_replication.py\", line 222, in test_data_replication\n    results = await get_max_written_value(first_model, second_model)\n  File \"/home/runner/work/mysql-k8s-operator/mysql-k8s-operator/tests/integration/high_availability/test_async_replication.py\", line 420, in get_max_written_value\n    values = await execute_queries_on_unit(\n  File \"/home/runner/work/mysql-k8s-operator/mysql-k8s-operator/tests/integration/helpers.py\", line 188, in execute_queries_on_unit\n    cursor.execute(query)\n  File \"/home/runner/work/mysql-k8s-operator/mysql-k8s-operator/.tox/integration/lib/python3.10/site-packages/mysql/connector/cursor_cext.py\", line 330, in execute\n    result = self._cnx.cmd_query(\n  File \"/home/runner/work/mysql-k8s-operator/mysql-k8s-operator/.tox/integration/lib/python3.10/site-packages/mysql/connector/connection_cext.py\", line 619, in cmd_query\n    raise get_mysql_exception(\nmysql.connector.errors.ProgrammingError: 1049 (42000): Unknown database 'continuous_writes_database'","flaky":false,"newFailed":false,"newBroken":true,"newPassed":false,"retriesCount":1,"retriesStatusChange":true,"beforeStages":[{"name":"tmp_path_factory","time":{"start":1719019734231,"stop":1719019734231,"duration":0},"status":"passed","steps":[],"attachments":[],"parameters":[],"stepsCount":0,"shouldDisplayMessage":false,"attachmentsCount":0,"hasContent":false,"attachmentStep":false},{"name":"ops_test","time":{"start":1719019734231,"stop":1719019734537,"duration":306},"status":"passed","steps":[],"attachments":[],"parameters":[],"stepsCount":0,"shouldDisplayMessage":false,"attachmentsCount":0,"hasContent":false,"attachmentStep":false},{"name":"event_loop","time":{"start":1719019734231,"stop":1719019734231,"duration":0},"status":"passed","steps":[],"attachments":[],"parameters":[],"stepsCount":0,"shouldDisplayMessage":false,"attachmentsCount":0,"hasContent":false,"attachmentStep":false},{"name":"second_model","time":{"start":1719019734537,"stop":1719019734881,"duration":344},"status":"passed","steps":[],"attachments":[],"parameters":[],"stepsCount":0,"shouldDisplayMessage":false,"attachmentsCount":0,"hasContent":false,"attachmentStep":false},{"name":"first_model","time":{"start":1719019734537,"stop":1719019734537,"duration":0},"status":"passed","steps":[],"attachments":[],"parameters":[],"stepsCount":0,"shouldDisplayMessage":false,"attachmentsCount":0,"hasContent":false,"attachmentStep":false},{"name":"continuous_writes","time":{"start":1719020052183,"stop":1719020054222,"duration":2039},"status":"passed","steps":[],"attachments":[],"parameters":[],"stepsCount":0,"shouldDisplayMessage":false,"attachmentsCount":0,"hasContent":false,"attachmentStep":false},{"name":"abort_on_fail","time":{"start":1719020052183,"stop":1719020052183,"duration":0},"status":"passed","steps":[],"attachments":[],"parameters":[],"stepsCount":0,"shouldDisplayMessage":false,"attachmentsCount":0,"hasContent":false,"attachmentStep":false}],"testStage":{"description":"Test to write to primary, and read the same data back from replicas.","status":"broken","statusMessage":"mysql.connector.errors.ProgrammingError: 1049 (42000): Unknown database 'continuous_writes_database'","statusTrace":"Traceback (most recent call last):\n  File \"/home/runner/work/mysql-k8s-operator/mysql-k8s-operator/.tox/integration/lib/python3.10/site-packages/mysql/connector/connection_cext.py\", line 611, in cmd_query\n    self._cmysql.query(\n_mysql_connector.MySQLInterfaceError: Unknown database 'continuous_writes_database'\n\nThe above exception was the direct cause of the following exception:\n\nTraceback (most recent call last):\n  File \"/home/runner/work/mysql-k8s-operator/mysql-k8s-operator/.tox/integration/lib/python3.10/site-packages/_pytest/runner.py\", line 341, in from_call\n    result: Optional[TResult] = func()\n  File \"/home/runner/work/mysql-k8s-operator/mysql-k8s-operator/.tox/integration/lib/python3.10/site-packages/_pytest/runner.py\", line 262, in <lambda>\n    lambda: ihook(item=item, **kwds), when=when, reraise=reraise\n  File \"/home/runner/work/mysql-k8s-operator/mysql-k8s-operator/.tox/integration/lib/python3.10/site-packages/pluggy/_hooks.py\", line 513, in __call__\n    return self._hookexec(self.name, self._hookimpls.copy(), kwargs, firstresult)\n  File \"/home/runner/work/mysql-k8s-operator/mysql-k8s-operator/.tox/integration/lib/python3.10/site-packages/pluggy/_manager.py\", line 120, in _hookexec\n    return self._inner_hookexec(hook_name, methods, kwargs, firstresult)\n  File \"/home/runner/work/mysql-k8s-operator/mysql-k8s-operator/.tox/integration/lib/python3.10/site-packages/pluggy/_callers.py\", line 182, in _multicall\n    return outcome.get_result()\n  File \"/home/runner/work/mysql-k8s-operator/mysql-k8s-operator/.tox/integration/lib/python3.10/site-packages/pluggy/_result.py\", line 100, in get_result\n    raise exc.with_traceback(exc.__traceback__)\n  File \"/home/runner/work/mysql-k8s-operator/mysql-k8s-operator/.tox/integration/lib/python3.10/site-packages/pluggy/_callers.py\", line 103, in _multicall\n    res = hook_impl.function(*args)\n  File \"/home/runner/work/mysql-k8s-operator/mysql-k8s-operator/.tox/integration/lib/python3.10/site-packages/_pytest/runner.py\", line 177, in pytest_runtest_call\n    raise e\n  File \"/home/runner/work/mysql-k8s-operator/mysql-k8s-operator/.tox/integration/lib/python3.10/site-packages/_pytest/runner.py\", line 169, in pytest_runtest_call\n    item.runtest()\n  File \"/home/runner/work/mysql-k8s-operator/mysql-k8s-operator/.tox/integration/lib/python3.10/site-packages/_pytest/python.py\", line 1792, in runtest\n    self.ihook.pytest_pyfunc_call(pyfuncitem=self)\n  File \"/home/runner/work/mysql-k8s-operator/mysql-k8s-operator/.tox/integration/lib/python3.10/site-packages/pluggy/_hooks.py\", line 513, in __call__\n    return self._hookexec(self.name, self._hookimpls.copy(), kwargs, firstresult)\n  File \"/home/runner/work/mysql-k8s-operator/mysql-k8s-operator/.tox/integration/lib/python3.10/site-packages/pluggy/_manager.py\", line 120, in _hookexec\n    return self._inner_hookexec(hook_name, methods, kwargs, firstresult)\n  File \"/home/runner/work/mysql-k8s-operator/mysql-k8s-operator/.tox/integration/lib/python3.10/site-packages/pluggy/_callers.py\", line 182, in _multicall\n    return outcome.get_result()\n  File \"/home/runner/work/mysql-k8s-operator/mysql-k8s-operator/.tox/integration/lib/python3.10/site-packages/pluggy/_result.py\", line 100, in get_result\n    raise exc.with_traceback(exc.__traceback__)\n  File \"/home/runner/work/mysql-k8s-operator/mysql-k8s-operator/.tox/integration/lib/python3.10/site-packages/pluggy/_callers.py\", line 103, in _multicall\n    res = hook_impl.function(*args)\n  File \"/home/runner/work/mysql-k8s-operator/mysql-k8s-operator/.tox/integration/lib/python3.10/site-packages/_pytest/python.py\", line 194, in pytest_pyfunc_call\n    result = testfunction(**testargs)\n  File \"/home/runner/work/mysql-k8s-operator/mysql-k8s-operator/.tox/integration/lib/python3.10/site-packages/pytest_asyncio/plugin.py\", line 529, in inner\n    _loop.run_until_complete(task)\n  File \"/usr/lib/python3.10/asyncio/base_events.py\", line 649, in run_until_complete\n    return future.result()\n  File \"/home/runner/work/mysql-k8s-operator/mysql-k8s-operator/tests/integration/high_availability/test_async_replication.py\", line 222, in test_data_replication\n    results = await get_max_written_value(first_model, second_model)\n  File \"/home/runner/work/mysql-k8s-operator/mysql-k8s-operator/tests/integration/high_availability/test_async_replication.py\", line 420, in get_max_written_value\n    values = await execute_queries_on_unit(\n  File \"/home/runner/work/mysql-k8s-operator/mysql-k8s-operator/tests/integration/helpers.py\", line 188, in execute_queries_on_unit\n    cursor.execute(query)\n  File \"/home/runner/work/mysql-k8s-operator/mysql-k8s-operator/.tox/integration/lib/python3.10/site-packages/mysql/connector/cursor_cext.py\", line 330, in execute\n    result = self._cnx.cmd_query(\n  File \"/home/runner/work/mysql-k8s-operator/mysql-k8s-operator/.tox/integration/lib/python3.10/site-packages/mysql/connector/connection_cext.py\", line 619, in cmd_query\n    raise get_mysql_exception(\nmysql.connector.errors.ProgrammingError: 1049 (42000): Unknown database 'continuous_writes_database'","steps":[],"attachments":[{"uid":"8442fb320adfaa9e","name":"log","source":"8442fb320adfaa9e.txt","type":"text/plain","size":713}],"parameters":[],"stepsCount":0,"shouldDisplayMessage":true,"attachmentsCount":1,"hasContent":true,"attachmentStep":false},"afterStages":[{"name":"continuous_writes::finalizer","time":{"start":1719020062348,"stop":1719020063369,"duration":1021},"status":"passed","steps":[],"attachments":[],"parameters":[],"stepsCount":0,"shouldDisplayMessage":false,"attachmentsCount":0,"hasContent":false,"attachmentStep":false},{"name":"abort_on_fail::0","time":{"start":1719020063369,"stop":1719020063369,"duration":0},"status":"passed","steps":[],"attachments":[],"parameters":[],"stepsCount":0,"shouldDisplayMessage":false,"attachmentsCount":0,"hasContent":false,"attachmentStep":false},{"name":"second_model::finalizer","time":{"start":1719020063380,"stop":1719020063395,"duration":15},"status":"passed","steps":[],"attachments":[],"parameters":[],"stepsCount":0,"shouldDisplayMessage":false,"attachmentsCount":0,"hasContent":false,"attachmentStep":false},{"name":"ops_test::finalizer","time":{"start":1719020063396,"stop":1719020064208,"duration":812},"status":"passed","steps":[],"attachments":[],"parameters":[],"stepsCount":0,"shouldDisplayMessage":false,"attachmentsCount":0,"hasContent":false,"attachmentStep":false},{"name":"event_loop::2","time":{"start":1719020064209,"stop":1719020064209,"duration":0},"status":"passed","steps":[],"attachments":[],"parameters":[],"stepsCount":0,"shouldDisplayMessage":false,"attachmentsCount":0,"hasContent":false,"attachmentStep":false},{"name":"event_loop::_close_event_loop","time":{"start":1719020064209,"stop":1719020064209,"duration":0},"status":"passed","steps":[],"attachments":[],"parameters":[],"stepsCount":0,"shouldDisplayMessage":false,"attachmentsCount":0,"hasContent":false,"attachmentStep":false},{"name":"event_loop::_provide_clean_event_loop","time":{"start":1719020064209,"stop":1719020064210,"duration":1},"status":"passed","steps":[],"attachments":[],"parameters":[],"stepsCount":0,"shouldDisplayMessage":false,"attachmentsCount":0,"hasContent":false,"attachmentStep":false}],"labels":[{"name":"tag","value":"group(1)"},{"name":"tag","value":"abort_on_fail"},{"name":"tag","value":"@pytest.mark.skipif(False, reason='Requires juju 3+')"},{"name":"tag","value":"asyncio"},{"name":"parentSuite","value":"tests.integration.high_availability"},{"name":"suite","value":"test_async_replication"},{"name":"host","value":"fv-az1426-41"},{"name":"thread","value":"12653-MainThread"},{"name":"framework","value":"pytest"},{"name":"language","value":"cpython3"},{"name":"package","value":"tests.integration.high_availability.test_async_replication"},{"name":"resultFormat","value":"allure2"}],"parameters":[],"links":[],"hidden":false,"retry":false,"extra":{"severity":"normal","retries":[{"uid":"8b191a492e53d56d","status":"unknown","time":{}}],"categories":[{"name":"Test defects","matchedStatuses":[],"flaky":false}],"history":{"statistic":{"failed":0,"broken":1,"skipped":12,"passed":53,"unknown":0,"total":66},"items":[{"uid":"5e16decfb3160bc0","reportUrl":"../1990//#testresult/5e16decfb3160bc0","status":"skipped","statusDetails":"XFAIL reason: aborted\n\n_pytest.outcomes.XFailed: aborted","time":{"start":1718933331424,"stop":1718933331424,"duration":0}},{"uid":"39389862f89c1300","reportUrl":"../1984//#testresult/39389862f89c1300","status":"passed","time":{"start":1718847365709,"stop":1718847373943,"duration":8234}},{"uid":"f0f59bfca00d344","reportUrl":"../1981//#testresult/f0f59bfca00d344","status":"passed","time":{"start":1718760938815,"stop":1718760947030,"duration":8215}},{"uid":"4cca65597e5d771a","reportUrl":"../1979//#testresult/4cca65597e5d771a","status":"passed","time":{"start":1718674538067,"stop":1718674546314,"duration":8247}},{"uid":"5d0407f61ae1a2b6","reportUrl":"../1978//#testresult/5d0407f61ae1a2b6","status":"passed","time":{"start":1718588198709,"stop":1718588206937,"duration":8228}},{"uid":"d2f8396c5311c573","reportUrl":"../1977//#testresult/d2f8396c5311c573","status":"passed","time":{"start":1718501800448,"stop":1718501808664,"duration":8216}},{"uid":"ce29748d18bfc813","reportUrl":"../1976//#testresult/ce29748d18bfc813","status":"passed","time":{"start":1718415354688,"stop":1718415363002,"duration":8314}},{"uid":"fe877de386b26c5a","reportUrl":"../1971//#testresult/fe877de386b26c5a","status":"passed","time":{"start":1718328937522,"stop":1718328945761,"duration":8239}},{"uid":"8cf84b02dbc0ad1","reportUrl":"../1964//#testresult/8cf84b02dbc0ad1","status":"skipped","statusDetails":"Skipped: async def function and no async plugin installed (see warnings)","time":{"start":1718242193351,"stop":1718242193351,"duration":0}},{"uid":"99a33018700a4e7e","reportUrl":"../1953//#testresult/99a33018700a4e7e","status":"skipped","statusDetails":"Skipped: async def function and no async plugin installed (see warnings)","time":{"start":1718155780240,"stop":1718155780241,"duration":1}},{"uid":"f654f160d05d4272","reportUrl":"../1950//#testresult/f654f160d05d4272","status":"skipped","statusDetails":"Skipped: async def function and no async plugin installed (see warnings)","time":{"start":1718069393801,"stop":1718069393802,"duration":1}},{"uid":"4387177022904b86","reportUrl":"../1943//#testresult/4387177022904b86","status":"skipped","statusDetails":"Skipped: async def function and no async plugin installed (see warnings)","time":{"start":1717982894236,"stop":1717982894236,"duration":0}},{"uid":"7e1a502384c41e81","reportUrl":"../1942//#testresult/7e1a502384c41e81","status":"skipped","statusDetails":"Skipped: async def function and no async plugin installed (see warnings)","time":{"start":1717896667119,"stop":1717896667119,"duration":0}},{"uid":"f83c453f22255eb7","reportUrl":"../1931//#testresult/f83c453f22255eb7","status":"passed","time":{"start":1717724165199,"stop":1717724173432,"duration":8233}},{"uid":"ad1f62082e2fa00c","reportUrl":"../1928//#testresult/ad1f62082e2fa00c","status":"passed","time":{"start":1717638067077,"stop":1717638075291,"duration":8214}},{"uid":"783fd1b3e1f970d5","reportUrl":"../1923//#testresult/783fd1b3e1f970d5","status":"passed","time":{"start":1717551336091,"stop":1717551344319,"duration":8228}},{"uid":"bc0b1ea3fb1b2b47","reportUrl":"../1918//#testresult/bc0b1ea3fb1b2b47","status":"passed","time":{"start":1717465349543,"stop":1717465357763,"duration":8220}},{"uid":"7785843fdf9df5ef","reportUrl":"../1917//#testresult/7785843fdf9df5ef","status":"passed","time":{"start":1717378714526,"stop":1717378722740,"duration":8214}},{"uid":"c9d1f6d2f32f2929","reportUrl":"../1916//#testresult/c9d1f6d2f32f2929","status":"passed","time":{"start":1717292184806,"stop":1717292193049,"duration":8243}},{"uid":"88a0b60796297447","reportUrl":"../1915//#testresult/88a0b60796297447","status":"passed","time":{"start":1717205828034,"stop":1717205836257,"duration":8223}}]},"tags":["asyncio","abort_on_fail","@pytest.mark.skipif(False, reason='Requires juju 3+')","group(1)"]},"source":"4c4e2ed4c9924b5b.json","parameterValues":[]}